examples/main/main.cpp

```diff
@@ -628,7 +634,23 @@ bool output_json(struct whisper_context * ctx, const char * fname, const whisper
                         value_i("from", t0 * 10, false);
                         value_i("to", t1 * 10, true);
                     end_obj(false);
-                    value_s("text", text, !params.diarize && !params.tinydiarize);
+                    value_s("text", text, !params.diarize && !params.tinydiarize && !token_timing);
+
+                    if (token_timing) {
+                        start_arr("tokens");
+                        const int n = whisper_full_n_tokens(ctx, i);
+                        for (int j = 0; j < n; ++j) {
+                            auto token = whisper_full_get_token_data(ctx, i, j);
+                            start_obj(nullptr);
+                                value_s("text", whisper_token_to_str(ctx, token.id), false);
+                                value_i("from", token.t0 * 10, false);
+                                value_i("to", token.t1 * 10, false);
+                                value_i("id", token.id, false);
+                                value_i("p", token.p, true);
```

Comment:
`value_i` accepts only integers as input, while `token.p` is a `float`. Therefore, to write a `float` into JSON, you'll need to create a new function named `value_f`.

https://github.com/ggerganov/whisper.cpp/blob/6b4f5f4488f0575db69c2d1de4f4009d4f73f5fd/whisper.h#L74-L90